test/pkg/environment/aws/expectations.go

```diff
@@ -362,6 +366,41 @@ func (env *Environment) GetAMIBySSMPath(ssmPath string) string {
 	return *parameter.Parameter.Value
 }
 
+// GetSSMPath gets the SSM path for an AMI release specified by the family, k8s version, architecture, and release offset.
+// A release offset of 0 will get you the latest release, 1 the previous release, and so on.
+// Note: this is tighly coupled with the naming conventions for AL2 and AL2023. This is **NOT** compatible with Bottlerocket, Ubuntu, or Windows AMIs.
```

Comment:
Why does it have to be tightly coupled? For any given family, shoudn't we be able to get its relative path and then use that to be the list of all the amis published to it, ordering them by date. From what I remember, you can definitely check when the parameter was published and use that as a proxy for determining which one comes before the other one

Reply:
The update timestamps for the SSM paths didn't line up with the release date, e.g. an SSM path may have been updated after it's release. I can dig in a little more to see if there's a better option here, but this seemed like a reasonable compromise to keep the code simple considering that 99% of the time in our tests we're using AL2023.